core/unit_test/TestRealloc.hpp

```diff
@@ -23,6 +23,9 @@ namespace TestViewRealloc {
 
 struct Default {};
 struct WithoutInitializing {};
+struct NoDefaultConstructor {
+  NoDefaultConstructor(int) {}
+};
```

Comment:
The two guys above are tags and you define something that is meant to be used as a value type in a view.
It is confusing to collocate them.
I suspect you can definite it within the block you added which IMO will make things more readable.

Reply:
Also I am not sure it is a good idea to have an empty type because that's probably not what you intend to test here.  Please add an int data member.